tedana/metrics/dependence.py

```diff
@@ -0,0 +1,354 @@
+"""
+Fit models.
+"""
+import logging
+import os.path as op
+
+import numpy as np
+import pandas as pd
+from scipy import stats
+
+from tedana import io, utils
+from tedana.stats import getfbounds, computefeats2, get_coeffs, t_to_z
+
+
+LGR = logging.getLogger(__name__)
+RepLGR = logging.getLogger('REPORT')
+RefLGR = logging.getLogger('REFERENCES')
+
+
+def calculate_weights(data_optcom, mixing_z):
+    """
+    Calculate weights.
+
+    Parameters
+    ----------
+    data_optcom : (S x T) array_like
+        Optimally combined data
+    mixing_z : (T x C) array_like
+        Z-scored mixing matrix
+
+    Returns
+    -------
+    weights : (S x C) array_like
+        Parameter estimates for optimally combined data against the mixing
+        matrix.
+    """
+    # compute un-normalized weight dataset (features)
+    weights = computefeats2(data_optcom, mixing_z, normalize=False)
+    return weights
+
+
+def calculate_betas(data_optcom, mixing):
+    assert data_optcom.shape[1] == mixing.shape[0]
+    # demean optimal combination
+    data_optcom_dm = data_optcom - data_optcom.mean(axis=-1, keepdims=True)
+    # compute PSC dataset - shouldn't have to refit data
+    betas = get_coeffs(data_optcom_dm, mixing)
+    return betas
+
+
+def calculate_psc(data_optcom, optcom_betas):
+    assert data_optcom.shape[1] == optcom_betas.shape[0]
+    PSC = 100 * optcom_betas / data_optcom.mean(axis=-1, keepdims=True)
+    return PSC
+
+
+def compute_countsigFT2(F_T2_clmaps):
+    countsigFT2 = F_T2_clmaps.sum(axis=0)
+    return countsigFT2
+
+
+def compute_countsigFS0(F_S0_clmaps):
+    countsigFS0 = F_S0_clmaps.sum(axis=0)
+    return countsigFS0
+
+
+def compute_countsignal(Z_clmaps):
+    countsignal = Z_clmaps.sum(axis=0)
+    return countsignal
```

Comment:
These three functions here are all doing the same thing, but just operating on different arrays. Can we make one function (e.g., `compute_countsig()`) instead, so that we can reduce having to test/maintain all three?